### PR TITLE
fix(line): update type for custom line canvas layers

### LINKED ISSUE #1413: Custom line canvas layers have different props then custom svg line layers
A custom canvas line layer has these arguments:

https://github.com/plouc/nivo/blob/ce7c755b23cfc55ead9bac9920a55ef6e764bd99/packages/line/src/LineCanvas.js#L122-L135

But uses this type definition:

https://github.com/plouc/nivo/blob/ce7c755b23cfc55ead9bac9920a55ef6e764bd99/packages/line/index.d.ts#L68-L76

Rather then inheriting its layer type definition form line props a specific type definition needs to be added.

https://github.com/plouc/nivo/blob/ce7c755b23cfc55ead9bac9920a55ef6e764bd99/packages/line/index.d.ts#L216-L218

### PR DESCRIPTION
**Pull request by jguddas**
this resolves #1413

### CODE REVIEW

File: packages/line/index.d.ts [hunk 1]
```diff
@@ -213,8 +213,17 @@ declare module '@nivo/line' {
     export class Line extends React.Component<LineSvgProps & Dimensions> {}
     export class ResponsiveLine extends React.Component<LineSvgProps> {}
 
+    export interface CustomCanvasLayerProps extends CustomLayerProps {
+        ctx: CanvasRenderingContext2D
+        lineGenerator: any
```
Comment (wyze) [inline packages/line/index.d.ts:218]: Is there a reason you are overriding this to `any`?

Comment (jguddas) [reply packages/line/index.d.ts]: @wyze I did a bit more testing and added:

```typescript
    export interface LineGenerator {
        (data: Datum[]): string
        x:any
        y:any
        defined:any
        curve:any
        context:any
    }
```

I don't know what the correct typings for `x`, `y`, `defined`, `curve` and `context` are I just know that they where missing.

Comment (wyze) [reply packages/line/index.d.ts]: No worries, just pushed a fix. :)

**Comment by wyze** [comment]
Thanks @jguddas!

**Comment by jguddas** [comment]
Awesome, thanks for the help and the quick me merge @wyze

**Pull request merged**